### PR TITLE
Feature/deletion policies

### LINKED ISSUE #1867: Implement Deletion Policies across providers
`DeletionPolicy` is a feature that depends on the provider to return a `SecretNotFound` error whenever the operation failed because the secret didn't yet exist.

This needs to be implemented for the following providers:

- [ ] #1868
- [ ] #1869
- [ ] #1870
- [x] #1871

### LINKED ISSUE #1868: Implement SecretNotFound returns for GCP Secret Manager
Whenever the client fails to get a secret because it was not found, a specific error `SecretNotFound` should be output. This allows `deletionPolicy` compatibility for GCP Secrets Manager.

### LINKED ISSUE #1869: Implement SecretNotFound returns for Azure Key vault
Whenever the client fails to get a secret because it was not found, a specific error `SecretNotFound` should be output. This allows `deletionPolicy` compatibility for Azure Keyvault.

### LINKED ISSUE #1870: Implement SecretNotFound returns for Kubernetes
Whenever the client fails to get a secret because it was not found, a specific error `SecretNotFound` should be output. This allows `deletionPolicy` compatibility for Kubernetes.

### PR DESCRIPTION
**Pull request by gusfcarvalho**
Fixes #1867 
Fixes #1868
Fixes #1869
Fixes #1870

### CODE REVIEW

File: docs/introduction/stability-support.md [hunk 1]
```diff
@@ -44,24 +44,24 @@ The following table describes the stability level of each provider and who's res
...
-| 1Password                 |      x       |              |                      |                         |        x         |             |
-| Generic Webhook           |              |              |                      |                         |                  |             |
-| senhasegura DSM           |              |              |                      |                         |        x         |             |
-| Doppler                   |      x       |              |                      |                         |        x         |             |
+| Provider                  | find by name | find by tags | metadataPolicy Fetch | referent authentication | store validation | push secret | DeletionPolicy Merge/Delete |
+|---------------------------|:------------:|:------------:| :------------------: | :---------------------: | :--------------: | :---------: | :-------------------------: 
+| AWS Secrets Manager       |      x       |      x       |                      |            x            |        x         |             |            x                |
```
[the remark this answers is not in the document]
Comment (gusfcarvalho) [reply docs/introduction/stability-support.md:49]: Plain dumb mistake! :D thanks for the catch!

**Review by moolen — APPROVED** [review]
LGTM, glad to see we're moving fast towards feature-completeness :+1:

**Comment by knelasevero** [comment]
🎉

**Pull request merged**